.github/workflows/ci-build.yml

```diff
@@ -1,76 +1,95 @@
 name: CI Build
 
 on:
-  - push
+  push:
+    branches:
+      - main
+  pull_request:
 
 jobs:
   build:
-    name: Build with Scala ${{ matrix.scala }}, Spark ${{ matrix.spark }} and Exasol ${{ matrix.exasol-docker-version }}
+    name: Building with ${{ matrix.mvn.name }} and Exasol ${{ matrix.exasol-docker-version }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        scala: [ 2.12.14 ]
-        spark: [ 3.0.2, 3.1.2 ]
-        exasol-docker-version: [ 6.2.16-d1, 7.0.12, 7.1.0-d1 ]
-
+        exasol-docker-version: [ 7.0.14, 7.1.3 ]
+        mvn:
```

Comment:
Do you really need to test all combinations (exasol-docker-version x mvn)? Would it maybe be sufficient to test only one combination with the old Exasol and the others with the new?

Reply:
I need all spark version since they are the ones used in most of customers I observed. But Exasol versions for sure can be for the last one, so it will +1 to the spark matrix.